subprojects/core/src/test/groovy/org/gradle/api/internal/file/collections/jdk7/Jdk7DirectoryWalkerTest.groovy

```diff
@@ -319,4 +321,30 @@ class Jdk7DirectoryWalkerTest extends Specification {
         }
     }
 
+    @Requires(TestPrecondition.FILE_PERMISSIONS)
+    @Issue('https://github.com/gradle/gradle/issues/2639')
+    @Ignore
+    def "excluded files' permissions should be ignored"() {
+        given:
+        def rootDir = tmpDir.createDir('root')
+        def unauthorizedDir = rootDir.createDir('unauthorized')
+        def permissionHandler = new PosixJdk7FilePermissionHandler()
+        permissionHandler.chmod(unauthorizedDir, 0222)
+        rootDir.createDir('authorized')
+
+        when:
+        def walkerInstance = new Jdk7DirectoryWalker()
+        def fileTree = new DirectoryFileTree(rootDir, new PatternSet().exclude('unauthorized'), {
```

Comment:
:o: It would be nice to also test that it is still failing in case of unauthorized files that are not excluded. It could be the same test as this one just without `exclude('unauthorized')` and expecting an `AccessDeniedException` exception. That would test that the _pattern set_ is respected and that not just all unauthorized files are ignored.

Reply:
Good. This scenario is put into an extra [test file](https://github.com/blindpirate/gradle/blob/8421a4f72979cd730c8d6e10e8e96921d18ca49d/subprojects/core/src/test/groovy/org/gradle/api/internal/file/collections/jdk7/Jdk7UnauthorizedDirectoryWalkerTest.groovy).